test/karma-testharness-adapter.js

```diff
@@ -122,9 +122,29 @@
     return true;
   }
 
-  function runRemainingTests(remainingTestURLs, config, testNameDiv, iframe) {
+  // Serialises the failures suitable for pasting into expectedFailures: {} in web-platform-tests-expectations.js
+  function formatFailures(failures) {
+    var testURLs = Object.keys(failures);
+    testURLs.sort();
+    return testURLs.map(function(testURL) {
+      var tests = Object.keys(failures[testURL]);
+      tests.sort();
+      return (
```

Comment:
Could you use JSON.stringify and then a string replace for " to '? That seems more robust/easier to read.